tests/test_client.py

```diff
@@ -426,8 +407,23 @@ def test_environment_check_gce_prod_with_working_gae_imports(self):
         with mock_module_import('google.appengine'):
             self._environment_check_gce_helper(status_ok=True)
 
-    def test_environment_check_gce_timeout(self):
-        self._environment_check_gce_helper(socket_error=True)
+    @mock.patch('oauth2client.client.os.environ',
+                new={client._SERVER_SOFTWARE: ''})
+    @mock.patch('oauth2client.transport.get_http_object',
+                return_value=object())
+    @mock.patch('oauth2client.transport.request',
+                side_effect=socket.timeout())
+    def test_environment_check_gce_timeout(self, mock_request, new_http):
+        self.assertFalse(client._in_gae_environment())
+        self.assertFalse(client._in_gce_environment())
+
+        # Verify mocks.
+        new_http.assert_called_once_with(timeout=client.GCE_METADATA_TIMEOUT)
+        headers = {
```

Comment:
Would use instead of the `test_client` module level constant, but the "if in `client`" discussion is still pending